xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsEventExecutor.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.internal;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.stream.IntStream;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.phase.Disposable;
+import org.xwiki.component.phase.Initializable;
+import org.xwiki.management.JMXBeanRegistration;
+import org.xwiki.mentions.internal.async.MentionsData;
+import org.xwiki.mentions.internal.jmx.JMXMentions;
+import org.xwiki.mentions.internal.jmx.JMXMentionsMBean;
+import org.xwiki.model.reference.DocumentReference;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+
+/**
+ * Default implementation of {@link MentionsEventExecutor}.
+ *
+ * This class is in charge of the management of mentions task analysis.
+ * First, when {@link DefaultMentionsEventExecutor#execute(DocumentReference, DocumentReference, String)} is called,
+ * a {@link MentionsData} is added to a queue.
+ * Then, a pool of {@link MentionsConsumer} workers is consuming the queue and delegate the actual analyis to
+ * {@link MentionsDataConsumer#consume(MentionsData)} which deals with actual implementation of the user mentions 
+ * analysis.  
+ *
+ * @version $Id$
+ * @since 12.6RC1
+ */
+@Component
+@Singleton
+public class DefaultMentionsEventExecutor implements MentionsEventExecutor, Initializable, Disposable
+{
+    private ThreadPoolExecutor executor;
```

Comment:
I don't really understand why you use an executor since you don't really use it as such. From what I understand all you do in the end is starting several persistent threads.

By the way you don't shutdown the executor or trigger a stop of the threads in dispose().